src/identify/grid.rs

```diff
@@ -345,7 +347,7 @@ fn jiggle_perspective<S>(img: &PreparedImage<S>, mut perspective: geometry::Pers
  * grid.
  */
 fn fitness_all<S>(img: &PreparedImage<S>, perspective: &geometry::Perspective, grid_size: usize) -> i32 where S: ImageBuffer {
-    let version = (grid_size - 17) / 4;
+    let version = ((grid_size - 17) / 4) % VERSION_DATA_BASE.len();
```

Comment:
Instead of papering over this issue: `grid_size` comes all the way from `SkewedGridLocation::from_group()`, where we can simply validate that it is within expected bounds and return None if it doesn't fit.